### ISSUE TITLE
System Classloader getting closed by the GroovyParser.clearCache() method

### ISSUE THREAD

**Issue by rbioteau** [groovy/groovy-eclipse#851]
When GroovyParser.clearCache() is called (when a project is closed or a Full build is invoked), all URLClassloader in the parent hierachy are closed including the System Classloader.

It leads to errors when the application try to load classes from the JRE like the crypto classes. 
The side effect is that it is no more possible to fetch a git repository by example, you get exceptions like
```
java.security.NoSuchAlgorithmException: SunTls12RsaPremasterSecret KeyGenerator not available
```

The GroovyClassLoaderFactory should filter the `Classloader.getSystemClassloader()` in the `close()` method.

**Comment by eric-milles** [comment]
Thanks for finding the source of this issue and reporting it.  I was wondering why I was seeing https problems in the past few days.

Ready to test.

**Comment by rbioteau** [comment]
The fix resolves my issue. 👍 
I don't know about the potential other side effects though.

Do you know if it is possible to integrate groovy-eclipse v3.3.0 on a 2018-12 target platform (with org.eclipse.jdt.core_3.16.0.xx-201812291959-e2018-12-RELEASE) ? Or do I need to update JDT too ?

**Comment by eric-milles** [comment]
Are you asking about the Java 12 patch or something else?

**Comment by rbioteau** [comment]
Not especially, it is just that we integrate groovy-eclipse in a RCP product ans we are targeting Eclipse 2018-12 target platform for our june release. The compatible version of groovy-eclipse for this version is v3.2.0. If I want this fix in my version for June, is it possible to keep all other plugins on the 2018-12 release train and only update groov-eclipse to v3.3.0 when it comes out ?
I known that groovy-eclipse patches jdt so there could some issues on this side of the integration.

**Comment by eric-milles** [comment]
This issue should not be present in the 3.2.0 release.  Use of system class loader was done in the past week as a response to Java 12 testing.  3.3.0 will be released very soon -- before the end of March.

Also, I'm curious about the product integration.  Can you share details about the product or is it not available to the public?  I always try to keep in mind the different circumstances where the compiler, editor and whatnot could be used when planning/making changes.

**Comment by rbioteau** [comment]
Sadly, this issue was present in 3.2.0. That's where I found it. As the close() was recusring all the parent URLClassloader, it was falling to the system class loader anyway.

I tried to integration the 3.3.0 snapshot and it looks ok.

I work at Bonitasoft. We use groovy as a scripting language in a BPM engine.
We have Studio which is an Eclipse RCP integrating groovy-eclipse. 
Bonita is open source if you want to have a look.  
https://fr.bonitasoft.com/ and https://github.com/bonitasoft/bonita-studio
